src/Transitioner.php

```diff
@@ -110,8 +111,28 @@ public function forwards(FormFlowInterface $flow): Status
         }
 
         $context = $flow->getContext();
-        $context->setCurrentStepNumber($currentStepNumber + 1);
-        $context->markCompleted($currentStep);
+        $context->setCompleted($currentStep);
+        $increment = 1;
+        if ($currentStepNumber < $flow->getSteps()->count()) {
+            $skipEvent = new SkipEvent($flow);
+            $this->dispatcher->dispatch($skipEvent, $this->createStepListenerId(FormFlowEvents::SKIP, $flow, $currentStepNumber));
+            $nextStep = $flow->getNextStep();
+            if ($skipEvent->isHardSkipped()) {
+                $context->setHardSkipped($nextStep);
+                ++$increment;
+            } else {
+                $context->unsetHardSkipped($nextStep);
+            }
+
+            if ($skipEvent->isSoftSkipped()) {
+                $context->setSoftSkipped($nextStep);
+                ++$increment;
+            } else {
+                $context->unsetSoftSkipped($nextStep);
```

Comment:
Is the array always filled ?

Reply:
no, but unset() is called directly inside the method.